plutus-wallet-api/ledger/Ledger/Scripts.hs

```diff
@@ -267,6 +270,19 @@ plcDataScriptHash = DataScriptHash . plcSHA2_256 . BSL.pack . BA.unpack
 plcValidatorDigest :: Digest SHA256 -> ValidatorHash
 plcValidatorDigest = ValidatorHash . BSL.pack . BA.unpack
 
+{-# INLINABLE plcAddress #-}
+-- | Get the SHA256 hash (for use in off-chain code) from a 'ValidatorHash'
+--   (on-chain)
+plcAddress :: ValidatorHash -> Maybe (Digest SHA256)
+plcAddress (ValidatorHash hsh) = digestFromByteString $ BSL.toStrict hsh
+
+{-# INLINABLE unsafePlcAddress #-}
+-- | Get the SHA256 hash (for use in off-chain code) from a 'ValidatorHash'
+--   (on-chain). Should be safe if 'ValidatorHash' was constructed using 
+--   'plcValidatorDigest' or 'plcValidatorHash'.
+unsafePlcAddress :: ValidatorHash -> Digest SHA256
+unsafePlcAddress = fromJust . plcAddress
```

Comment:
I've been meaning to do something about this for a while but: the mess of functions for dealing with scripts and their hashes is *very* confusing. Not sure what to do about it now but we should do something.

Reply:
I think we can get rid of need datascript and validator script hashes and the rest should just be `Address`